ivanti_security_controls/plugin.spec.yaml

```diff
@@ -0,0 +1,223 @@
+      title: Last Known IP Address
+      description: The last known IP address of the agent machine
+      type: string
+      required: false
+    listeningPort:
+      title: Listening Port
+      description: The listening port number
+      type: integer
+      required: false
+    machineName:
+      title: Machine Name
+      description: The agent machine's host name
+      type: string
+      required: false
+    reportedPolicyId:
+      title: Reported Policy ID
+      description: The agent policy ID
+      type: string
+      required: false
+    status:
+      title: Status
+      description: The current status of the agent
+      type: string
+      required: true
+    links:
+      title: Agent Links
+      description: Shows the related URLs for the agent
+      type: object
+      required: false
+  agent_status:
+    agentId:
+      title: Agent ID
+      description: The agent ID
+      type: string
+      required: true
+    frameworkVersion:
+      title: Framework Version
+      description: The installed agent framework version
+      type: object
+      required: false
+    installedPackages:
+      title: Installed Packages
+      description: The list of engines installed on the agent machine
+      type: "[]string"
+      required: false
+    lastCheckIn:
+      title: Last Check-In
+      description: The date and time of the most recent check-in
+      type: string
+      required: false
+    machineName:
+      title: Machine Name
+      description: The agent machine's host name
+      type: string
+      required: false
+    reportedOn:
+      title: Reported On
+      description: The time the information was gathered from the agent machine
+      type: string
+      required: false
+    runningPolicyId:
+      title: Running Policy ID
+      description: The agent's running policy ID
+      type: string
+      required: false
+    runningPolicyVersion:
+      title: Running Policy Version
+      description: The agent's policy ID
+      type: integer
+      required: false
+    links:
+      title: Agent Links
+      description: Shows the related URLs for the agent
+      type: object
+      required: false
+
+actions:
+  get_agent:
+    title: Get Agent
+    description: Retrieve Agent from Ivanti Security Controls
+    input:
+      id:
+        title: Agent ID
+        description: Agent Identifier
+        type: string
+        required: true
+        example: ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789ABCD
+    output:
+      agent:
+        title: Agent
+        description: Details about an agent
+        type: agent_detail
+        required: true
+  get_agents:
+    title: Get Agents
+    description: Retrieve Agent from Ivanti Security Controls
+    input:
+      listening_filter:
+        title: Listening Filter
+        description: Returns agents that are configured as listening agents
+        type: string
+        enum:
+        - "True"
+        - "False"
+        - ""
+        default: ""
```

Comment:
This can be empty? lol

Reply:
I didn't know it worked either 🤷

Reply:
Cleaned up input so it is a bit more user friendly.